### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by garciajordy**
Milestone-2:
-Add User class
-Add welcome statement
-Add user interface
-Add a loop to allocate turns to players

### CODE REVIEW

**Comment by mimipeshy** [comment]
### STATUS: CHANGES REQUIRED :writing_hand:

Hello team @garciajordy  and @JulianaOsemeke, congratulations on the effort so far in this milestone. This is a good start guys 👏🏾 

#### ToHighlight

- [ ] Good job on following the correct GitHub flow ✔️  
- [ ] Your commit messages are well written in the present tense ✔️ 
- [ ] Your PR description is short and clear ✔️ 

#### To Improve
The following moves still have to be implemented before approval

- [x] The game asks the player to select from available moves. [Example screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/select_play.png)
- [x] The game informs the player if the selected move is invalid (without computing what an invalid move is). [Example screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/invalid_move.png)
- [x] The game displays the board after the player move (without computing which spots on the board are taken). [Example screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/player_turn.png)
- [x] The game informs the player if the selected move is a winning move (without computing the winning move). [Example screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/game_win%2059.png)
- [x] The game informs the player if the selected move is a draw move (without computing draw). [Example screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/game_tie.png)
- [x] The game repeats all actions for the next player's move. [Example gif](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/game_flow_example.gif)

#### Implementation details

* [x] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))
* [x] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable:

```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```


That's all :smile: kindly submit another review when done.

_Feel free to reach out to me on slack if you have any questions or something is unclear, or just comment on this PR and tag my name  (@mimipeshy) and I will be glad to help_
Regards
Peris Ndanu

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: 
--------------
Hello, @JulianaOsemeke  @garciajordy  you did an amazing job on this milestone. Great job working on the required changes :muscle: 
However there a minor change to make to complete this milestone. :hammer_and_wrench: see it below :eyes: :point_down: 
----------------------

- [ ] You linters are not displaying in GitHub. Please fix it. Remember that before you send it for review the linters must be working on GitHub. Remember that `.github` folder starts with a dot.
---------------
Please fix this minor error and send it again for review :+1: :100: 
HAPPY CODING!!

CavernousEnergeticBluetonguelizard-size_restricted

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Status: Change Requested image



You did a good job so far Team 👏👏 but they say "Feedback is the breakfast of champions"
you still have small changes to make to be ready for the next adventures

- This branch you are working on (user_interface) does not have a README I was confused about how to play the game please add a decent README
- Fix small linters errors you have locally

# Happy Coding

best,

Lydia Ingabire

File: README.md
Comment (pacyL2K19) [inline README.md]: Nice to have a Readme file at this point, unfortunately, it didn't help me to play the game, kindly help the one who clones your repo to understand all the steps he needs to follow to make it work

File: bin/main
Comment (pacyL2K19) [inline bin/main]: You can ask users to put something simple like `Y or N` instead of `Yes or No` remember that users are lazy people

File: bin/main
Comment (pacyL2K19) [inline bin/main:65]: ### Highligth
Like this tip :+1:

**Review by pacyL2K19 — CHANGES_REQUESTED** [review]
# Change Required :stop_sign: 

Hello **Team** 
You made a great work working on previous code review changes requested and I enjoyed reading you code :blush: 
But before moving forward to the next milestone, there are some more changes you need to make
- [x] [Mandatory] There is a conflict in the branch, please resolve it so that we ca see the linter passing 
- [x] [Mandatory] The game doesn't play, it seems you have an unhandled exception in the code that doesn't allow users to continue playing the game, I was just able to enter the name but the rest didn't pass, please make sure users can play the game before submitting again 
warnTic
- [x] [Mandatory] Kindly work on the required changes bellow as additional ones 

Go ahead and work on these changes before you submit another PR and feel free to reach out in Slack or [LinkedIn](https://www.linkedin.com/in/pacifique-linjanja/) if you have any question 

### Happy coding :computer:

File: README.md
Comment (garciajordy) [reply README.md]: We add more specific instructions like requested

File: bin/main
Comment (garciajordy) [reply bin/main]: Changed "yes" to "y"

File: bin/main
Comment (garciajordy) [reply bin/main:65]: Hi Pacifique,

We tried to run the game and it works just perfectly.

Comment (pacyL2K19) [reply bin/main:65]: The readme file helped me to figure out

File: bin/main
Comment (pacyL2K19) [inline bin/main]: :+1:

**Review by pacyL2K19 — APPROVED** [review]
# Status: Approved
Hello Team, your code looks good, and I enjoyed playing your game and reading your code 
### Congratulations :tada: 
[Optional] In case you like colors, you can colorize your user interface thanks to the [Colorize](https://github.com/fazibear/colorize) gem which is really simple to use
Now go-ahead to the next milestone and don't forget to keep it high :smiley: 
Happy coding :desktop_computer:

**Pull request merged**
